platform/cog-platform-fdo.c

```diff
@@ -998,6 +987,11 @@ on_surface_frame (void *data, struct wl_callback *callback, uint32_t time)
         egl_data.eglDestroyImage (egl_data.display, wpe_view_data.image);
         wpe_view_data.image = NULL;
     }
+
+    if (wpe_view_data.buffer != NULL) {
+        wl_buffer_destroy (wpe_view_data.buffer);
+        wpe_view_data.buffer = NULL;
+    }
```

Comment:
Please replace this whole `if`-statement with [g_clear_pointer()](https://developer.gnome.org/glib/unstable/glib-Memory-Allocation.html#g-clear-pointer), which is equivalent:

```c
g_clear_pointer (&wpe_view_data.buffer, wl_buffer_destroy);
```

Most of the code in Cog uses the `g_clear_*` and `g_auto*` macros, and I think once one gets used to them, the code is more concise without being less explicit or unclear.